src/gui/sharedialog.cpp

```diff
@@ -327,13 +326,30 @@ void ShareDialog::slotCreateLinkShare()
     }
 }
 
+void ShareDialog::slotCreatePasswordForLinkShare(const QString &password)
+{
+    if (const auto shareLinkWidget = qobject_cast<ShareLinkWidget*>(sender())) {
```

Comment:
I'd assert on the sender, clearly this slot is not meant to be used as a regular method so better make that precondition explicit.